### PR TITLE
Add meta tags for discoverability into the HTML head tag

### PR DESCRIPTION
**Pull request by PenghaiZhang**
#1148 
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] screenshots are included showing significant UI changes

##### Description of change
With the New UI turned on,  meta tags for discoverability are missing and two test (`metaTest` and `validatationTest`) have been skipped in the New UI due to this issue.

This PR aims to fix this issue by passing the tags from back-end to front-end through `LegacyConetent` and then use React hook to dynamically control these tags.

Although there are no new tests against this fix, the two previously skipped tests are re enabled to work for these changes.

Screenshot from 2019-11-13 11-46-45
Screenshot from 2019-11-13 11-45-56

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/mainui/Template.tsx [hunk 1]
```diff
@@ -282,6 +288,29 @@ export const Template = React.memo(function Template(props: TemplateProps) {
...
   }, [props.title]);
 
+  React.useEffect(() => {
+    updateMetaTags(props.metaTags);
+  }, [props.metaTags]);
+
+  function updateMetaTags(tags: string | undefined) {
+    const head = document.head;
+    if (tags) {
+      // The meta tags generated on Server side separates each other by a newline symbol
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/mainui/Template.tsx]: Worded slightly awkwardly. How about:
```// The meta tags generated on the server side, separated by new line symbols```

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts [hunk 2]
```diff
@@ -370,5 +370,6 @@ export const languageStrings = {
       notor: "None match"
     },
     convertGroup: "Convert to group"
-  }
+  },
+  metatags: ["citation_title", "citation_description"]
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts]: Good morning Aaron. The intention of having these strings is just to pass to Selenium tests.

[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts]: So what about keeping a few general meta tags and remove others ? 
Screenshot from 2019-11-14 09-27-04

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/react.ftl [hunk 3]
```diff
@@ -5,7 +5,6 @@
     <head>
       <meta http-equiv="X-UA-Compatible" content="IE=edge" >
     	<meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
-      <meta name="robots" content="noindex, follow">
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/react.ftl]: Why remove this?

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts]: Yeah, I'm thinking we want to remove this - and then also the use in Template.tsx and the `else` block starting at Tempalte.tsx:303.

I'll give you a call to discuss.

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/react.ftl (hunk 3 above)
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/react.ftl]: Hi @edalex-ian , we added this tag in August in order to get tests pass with New UI turned on. Now due to the fix we can safely remove it.  This tag is actually controlled in `HarvestListSection` .

Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/react.ftl]: Excellent, thanks for the clarification. :+1:

**Pull request merged**